src/com/google/enterprise/adaptor/fs/WindowsFileDelegate.java

```diff
@@ -538,19 +581,26 @@ public void callback(int errorCode, int nBytesTransferred,
 
         // Signal any waiting threads that the monitor is now active.
         startSignal.countDown();
+        backOff.reset();
 
         boolean logging = !paused();
         if (logging) {
           log.log(Level.FINER, "Waiting for notifications for {0}.", watchPath);
         }
         int waitResult = kernel32.WaitForSingleObjectEx(stopEvent,
-            Kernel32.INFINITE, true);
+            15 * 60 * 1000 /* 15 min timeout in millisecs */, true);
```

Comment:
This is just to force getting a new handle in case we're in a limbo we haven't seen and don't expect? Is there an issue of missing notifications while the loop is restarted?